enclave/sgx/verifier.c

```diff
@@ -0,0 +1,308 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#include <openenclave/corelibc/stdio.h>
+#include <openenclave/enclave.h>
+#include <openenclave/internal/calls.h>
+#include <openenclave/internal/raise.h>
+#include <openenclave/internal/safecrt.h>
+#include <openenclave/internal/safemath.h>
+#include <openenclave/internal/crypto/sha.h>
+#include <openenclave/internal/report.h>
+#include <openenclave/internal/raise.h>
+#include <stdlib.h>
+#include <string.h>
+#include "platform_t.h"
+#include "verifier.h"
+
+//Hardcode Intel signed QvE Identity below
```

Comment:
We had discussion about QvE identity being in policy input or claim output. For step 1 this is hard-coded. This means any OE SDK release only works with one QvE {signer + prod id}. Any QvE release that changes prod id is a breaking change. Attestation apps may find themselves broken after Windows Update and the only way to resolve is to wait for the next OE release (hoping it has the right QvE prod id) and rebuild. **This is a major concern.**

Reply:
The hardcode info in this file, including Mrsigner, Attribute, Miscselect, Prod id, would never change (almost), the only QvE identity info would change is ISV SVN, so we suggest to add ISV SVN in policy input in step 2. Then OE SDK doesn't need to care about the version of QvE.

In step 1, I hardcode QvE ISV SVN threshold with `3` in function `oe_verify_quote_with_sgx_endorsements()`. In this case, it means only when QvE ISV SVN is smaller then 3, the function would return error, any updated QvE would pass. So it means  OE SDK doesn't need to care about the version of QvE also. 
But in step 1, when Intel update the QvE ISV SVN, there would be a potential security risk if OE SDK doesn't update the hardocde QvE ISV SVN threshold. For example, if latest QvE ISV SVN is 5, if we still using ISV SVN threshold 3, then QvE with ISV SVN 3, 4, 5 would pass the function.